lib/io.c

```diff
@@ -10,7 +10,7 @@
 #include <metal/sys.h>
 
 void metal_io_init(struct metal_io_region *io, void *virt,
-	      const metal_phys_addr_t *physmap, size_t size,
+	      const metal_phys_addr_t *physmap, size_t size, size_t offset,
```

Comment:
This changes a publicly facing API which we can't really do.

It's also incomplete, in that it doesn't actually use the offset anywhere. 

This is definitely a valid issue, but I believe it will require more thought and architecture to make libmetal use the offset properly. I think we can probably do something by adding an element to the libmetal Linux device structure and changing the way the memory region is initialized.

Reply:
I agree that this change cannot be taken as is, as explained it was only a way to initiate a talk.

> It's also incomplete, in that it doesn't actually use the offset anywhere.

Hum... maybe I've missed something, but please have a look at the metal_io_virt() function that is modified by the same patch where the new 'offset' field of the metal_io_region is used.

> This is definitely a valid issue, but I believe it will require more thought and architecture to make libmetal use the offset properly.

Should I make a new pull request with only the first patch that fix the issue + adding a note in the documentation saying that on Linux, devices that use memory that is not page aligned may not work?

Reply:
That would be much appreciated. 

We should also open an issue to track the problem of actually using the offset.

Reply:
@edmooring Done. Let me know what you think about the note that I've added into the README.md file

Reply:
Removed